price-feeder/config/currency_provider_tracker.go

```diff
@@ -0,0 +1,175 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+const (
+	coinGeckoRestHost        = "https://api.coingecko.com/"
+	coinGeckoRestPath        = "api/v3/coins/"
+	coinGeckoListEndpoint    = "list"
+	coinGeckoTickersEndpoint = "tickers"
+	trackingPeriod           = time.Hour * 24
+)
+
+type (
+	// CurrencyProviderTracker queries the CoinGecko API for all the exchanges that
+	// support the currency pairs set in the price feeder config. It will poll the API
+	// every 24 hours to log any new exchanges that were added for a given currency.
+	//
+	// REF: https://www.coingecko.com/en/api/documentation
+	CurrencyProviderTracker struct {
+		logger              zerolog.Logger
+		pairs               []CurrencyPair
+		coinIDSymbolMap     map[string]string   // ex: map["ATOM"] = "cosmos"
+		CurrencyProviders   map[string][]string // map of price feeder currencies and what exchanges support them
+		CurrencyProviderMin map[string]int      // map of price feeder currencies and min required providers for them
+	}
+
+	// List of assets on CoinGecko and their corresponding id and symbol.
+	CoinGeckoCoinList struct {
+		ID     string `json:"id"`     // ex: "cosmos"
+		Symbol string `json:"symbol"` // ex: "ATOM"
+	}
+
+	// CoinGecko ticker shows market data for a given currency pair including what
+	// exchanges they're on.
+	CoinGeckoCoinTickerResponse struct {
```

Comment:
Same here & for the others:
```suggestion
	coinTickerResponse struct {
```